### PR TITLE
Add `CEditorObject` class

### PR DESCRIPTION
**Pull request by Marmare314**
This class is slightly more light-weight than a `CEditorComponent` and its naming makes it more natural to inherit for envelope points, proof-mode positions etc.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/editor_object.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#ifndef GAME_EDITOR_EDITOR_OBJECT_H
+#define GAME_EDITOR_EDITOR_OBJECT_H
+
+#include <functional>
+#include <game/client/ui.h>
```
Comment (Robyt3) [inline src/game/editor/editor_object.h:5]: I think you only need to include `ui_rect.h` and forward-declare `CUI`.

Comment (Marmare314) [reply src/game/editor/editor_object.h]: `input.h` is also needed, but a lot better than `client.h`.

Comment (Marmare314) [reply src/game/editor/editor_object.h]: Is there any way to know why the mac build failed?

Comment (Robyt3) [reply src/game/editor/editor_object.h]: > Is there any way to know why the mac build failed?

fail

In this case I don't think so. Retry the CI by doing `git commit --amend` and `git push -f` and see if that fixes it.

Comment (Marmare314) [reply src/game/editor/editor_object.h]: Thanks, that worked!

Comment (Robyt3) [reply src/game/editor/editor_object.h]: Reloading the page also helped, now I can see the error log:

```
 In file included from /Users/runner/work/ddnet/ddnet/src/game/editor/component.cpp:1:
/Users/runner/work/ddnet/ddnet/src/game/editor/component.h:26:56: error: implicit instantiation of undefined template 'std::vector<std::reference_wrapper<CEditorComponent>>'
        std::vector<std::reference_wrapper<CEditorComponent>> m_vSubComponents = {};
                                                              ^
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/iosfwd:259:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
/Users/runner/work/ddnet/ddnet/src/game/editor/component.cpp:19:18: error: no matching member function for call to 'Editor'
                Component.Init(Editor());
                               ^~~~~~
/Users/runner/work/ddnet/ddnet/src/game/editor/editor_object.h:68:11: note: candidate function not viable: no known conversion from 'CEditorComponent' to 'CEditorObject' for object argument
        CEditor *Editor();
                 ^
/Users/runner/work/ddnet/ddnet/src/game/editor/editor_object.h:69:17: note: candidate function not viable: no known conversion from 'CEditorComponent' to 'const CEditorObject' for object argument
        const CEditor *Editor() const;
                       ^
2 errors generated.
```

Comment (Marmare314) [reply src/game/editor/editor_object.h]: Unfortunately I have no idea how that error can be fixed.

Comment (Robyt3) [reply src/game/editor/editor_object.h]: Probably means you are missing `#include <vector>`

Comment (Marmare314) [reply src/game/editor/editor_object.h]: Ah thats probably it, thanks!

**Review by Robyt3 — APPROVED** [review]

**Pull request merged**